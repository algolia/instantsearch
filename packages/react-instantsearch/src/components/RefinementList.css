.root {
}

.item {
}

.itemSelected {
}

.itemLabel {
}

.itemCount {
}

.showMore {
}

.showMoreDisabled {
}

.itemLabelSelected {
}

.itemCountSelected {
}

.itemCheckbox{
}

.itemCheckboxSelected{
}
