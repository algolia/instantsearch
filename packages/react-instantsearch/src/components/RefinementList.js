import React, {PropTypes, Component} from 'react';
import {pick} from 'lodash';
import translatable from '../core/translatable';
import List from './List';
import classNames from './classNames.js';
import Highlight from '../widgets/Highlight';
import SearchBox from '../components/SearchBox';
const cx = classNames('RefinementList');

class RefinementList extends Component {
  constructor(props) {
    super(props);
    this.state = {query: ''};
  }

  static propTypes = {
    translate: PropTypes.func.isRequired,
    refine: PropTypes.func.isRequired,
    searchForFacetValues: PropTypes.func,
    createURL: PropTypes.func.isRequired,
    items: PropTypes.arrayOf(PropTypes.shape({
      label: PropTypes.string.isRequired,
      value: PropTypes.arrayOf(PropTypes.string).isRequired,
      count: PropTypes.number.isRequired,
      isRefined: PropTypes.bool.isRequired,
    })),
    isFromSearch: PropTypes.bool.isRequired,
    showMore: PropTypes.bool,
    limitMin: PropTypes.number,
    limitMax: PropTypes.number,
    transformItems: PropTypes.func,
  };

  selectItem = item => {
    this.props.refine(item.value);
  };

  renderItem = item => {
    const label = item._highlightResult
      ? <Highlight attributeName="label" hit={item}/>
      : item.label;

    return (
      <label>
        <input
          {...cx('itemCheckbox', item.isRefined && 'itemCheckboxSelected')}
          type="checkbox"
          checked={item.isRefined}
          onChange={() => this.selectItem(item)}
        />
        <span {...cx('itemBox', 'itemBox', item.isRefined && 'itemBoxSelected')}></span>
        <span {...cx('itemLabel', 'itemLabel', item.isRefined && 'itemLabelSelected')}>
          {label}
        </span>
        {' '}
        <span {...cx('itemCount', item.isRefined && 'itemCountSelected')}>
          {item.count}
        </span>
      </label>);
  };

  render() {
    const facets = this.props.items.length > 0
      ? <List
        renderItem={this.renderItem}
        cx={cx}
        {...pick(this.props, [
          'translate',
          'items',
          'showMore',
          'limitMin',
          'limitMax',
        ])}
      />
      : <div>{this.props.translate('noResults')}</div>;

    const searchBox = this.props.searchForFacetValues ?
      <div {...cx('SearchBox')}>
        <SearchBox
          currentRefinement={this.props.isFromSearch ? this.state.query : ''}
          refine={value => {
            this.setState({query: value});
            this.props.searchForFacetValues(value);
          }}
          translate={this.props.translate}
          onSubmit={e => {
            e.preventDefault();
            e.stopPropagation();
            if (this.props.isFromSearch) {
              this.selectItem(this.props.items[0]);
            }
          }}
        />
      </div> : null;

    return (
      <div>
        {searchBox}
        {facets}
      </div>
    );
  }
}

export default translatable({
  showMore: extended => extended ? 'Show less' : 'Show more',
  noResults: 'No Results',
})(RefinementList);

