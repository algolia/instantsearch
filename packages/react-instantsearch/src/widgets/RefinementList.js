import connectRefinementList from '../connectors/connectRefinementList.js';
import RefinementListComponent from '../components/RefinementList.js';

/**
 * The RefinementList component displays a list that let the end user choose multiple values for a specific facet.
 * @name RefinementList
 * @kind component
 * @category widget
 * @propType {string} attributeName - the name of the attribute in the record
 * @propType {string} [operator=or] - How to apply the refinements. Possible values: 'or' or 'and'.
 * @propType {boolean} [showMore=false] - true if the component should display a button that will expand the number of items
 * @propType {number} [limitMin=10] - the minimum number of diplayed items
 * @propType {number} [limitMax=20] - the maximun number of displayed items. Only used when showMore is set to `true`
 * @propType {string[]} defaultRefinement - the values of the items selected by default
 * @themeKey ais-RefinementList__root - the root of the component
 * @themeKey ais-RefinementList__items - the container of all items in the list
 * @themeKey ais-RefinementList__item - a single item
 * @themeKey ais-RefinementList__itemSelected - the selected list item
 * @themeKey ais-RefinementList__itemCheckbox - the item checkbox
 * @themeKey ais-RefinementList__itemCheckboxSelected - the selected item checkbox
 * @themeKey ais-RefinementList__itemLabel - the item label
 * @themeKey ais-RefinementList__itemLabelSelected - the selected item label
 * @themeKey RefinementList__itemCount - the item count
 * @themeKey RefinementList__itemCountSelected - the selected item count
 * @themeKey ais-RefinementList__showMore - the button that let the user toggle more results
 * @translationkey showMore - The label of the show more button. Accepts one parameters, a boolean that is true if the values are expanded
 * @example
 * import React from 'react';
 *
 * import {RefinementList, InstantSearch} from '../packages/react-instantsearch/dom';
 *
 * export default function App() {
 *   return (
 *     <InstantSearch
 *       className="container-fluid"
 *       appId="latency"
 *       apiKey="6be0576ff61c053d5f9a3225e2a90f76"
 *       indexName="ikea"
 *     >
 *       <RefinementList attributeName="colors" />
 *     </InstantSearch>
 *   );
 * }
 */
export default connectRefinementList(RefinementListComponent);
