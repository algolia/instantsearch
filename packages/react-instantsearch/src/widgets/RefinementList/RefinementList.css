.root {
}

.item {
}

.item {
}

.itemSelected {
}

.itemLabel {
}

.itemCount {
}

.showMore {
}