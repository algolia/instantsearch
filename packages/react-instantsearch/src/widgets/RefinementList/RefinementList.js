import React, {PropTypes, Component} from 'react';
import pick from 'lodash/pick';

import themeable from '../../core/themeable';
import translatable from '../../core/translatable';

import List from '../../components/List';

import theme from './RefinementList.css';

class RefinementList extends Component {
  static propTypes = {
    applyTheme: PropTypes.func.isRequired,
    translate: PropTypes.func.isRequired,
    refine: PropTypes.func.isRequired,
    createURL: PropTypes.func.isRequired,
    items: PropTypes.arrayOf(PropTypes.shape({
      value: PropTypes.string.isRequired,
      count: PropTypes.number.isRequired,
    })),
    selectedItems: PropTypes.arrayOf(PropTypes.string),
    showMore: PropTypes.bool,
    limitMin: PropTypes.number,
    limitMax: PropTypes.number,
  };

  onItemChange = (item, e) => {
    const {selectedItems} = this.props;
    const nextSelectedItems = selectedItems.slice();
    const idx = nextSelectedItems.indexOf(item.value);
    if (e.target.checked && idx === -1) {
      nextSelectedItems.push(item.value);
    } else if (!e.target.checked && idx !== -1) {
      nextSelectedItems.splice(idx, 1);
    }
    this.props.refine(nextSelectedItems);
  }

  renderItem = (item, selected) => {
    const {translate, applyTheme} = this.props;

    return (
      <label>
        <input
          {...applyTheme('itemCheckbox', 'itemCheckbox')}
          type="checkbox"
          checked={selected}
          onChange={this.onItemChange.bind(null, item)}
        />
        <span {...applyTheme('itemLabel', 'itemLabel')}>
          {item.value}
        </span>
        {' '}
        <span {...applyTheme('itemCount', 'itemCount')}>
          {translate('count', item.count)}
        </span>
      </label>
    );
  };

  render() {
    return (
      <List
        renderItem={this.renderItem}
        {...pick(this.props, [
          'applyTheme',
          'translate',
          'items',
          'selectedItems',
          'showMore',
          'limitMin',
          'limitMax',
        ])}
      />
    );
  }
}

export default themeable(theme)(
  translatable({
    showMore: extended => extended ? 'Show less' : 'Show more',
    count: count => count.toLocaleString(),
  })(RefinementList)
);
