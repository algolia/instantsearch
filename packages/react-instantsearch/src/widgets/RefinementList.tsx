import React, { useRef, useState } from 'react';
import { useRefinementList } from 'react-instantsearch-core';

import { RefinementList as RefinementListUiComponent } from '../ui/RefinementList';
import { SearchBox as SearchBoxUiComponent } from '../ui/SearchBox';

import type { RefinementListProps as RefinementListUiComponentProps } from '../ui/RefinementList';
import type { SearchBoxTranslations } from '../ui/SearchBox';
import type { RefinementListItem } from 'instantsearch.js/es/connectors/refinement-list/connectRefinementList';
import type { RefinementListWidgetParams } from 'instantsearch.js/es/widgets/refinement-list/refinement-list';
import type { UseRefinementListProps } from 'react-instantsearch-core';

type UiProps = Pick<
  RefinementListUiComponentProps,
  | 'canRefine'
  | 'items'
  | 'onRefine'
  | 'query'
  | 'searchBox'
  | 'noResults'
  | 'canToggleShowMore'
  | 'onToggleShowMore'
  | 'isShowingMore'
  | 'translations'
>;

export type RefinementListProps = Omit<
  RefinementListUiComponentProps,
  keyof UiProps
> &
  UseRefinementListProps &
  Pick<RefinementListWidgetParams, 'searchable' | 'searchablePlaceholder'> & {
    translations?: Partial<
      UiProps['translations'] &
        SearchBoxTranslations & {
          /**
           * What to display when there are no results.
           */
          noResultsText: string;
        }
    >;
  };

export function RefinementList({
  searchable,
  searchablePlaceholder,
  attribute,
  operator,
  limit,
  showMore,
  showMoreLimit,
  sortBy,
  escapeFacetValues,
  transformItems,
  translations,
  ...props
}: RefinementListProps) {
  const {
    canRefine,
    canToggleShowMore,
    isFromSearch,
    isShowingMore,
    items,
    refine,
    searchForItems,
    toggleShowMore,
  } = useRefinementList(
    {
      attribute,
      operator,
      limit,
      showMore,
      showMoreLimit,
      sortBy,
      escapeFacetValues,
      transformItems,
    },
    {
      $$widgetType: 'ais.refinementList',
    }
  );
  const [inputValue, setInputValue] = useState('');
  const inputRef = useRef<HTMLInputElement>(null);

  function setQuery(newQuery: string) {
    setInputValue(newQuery);
    searchForItems(newQuery);
  }

  function onRefine(item: RefinementListItem) {
    refine(item.value);
    setQuery('');
  }

  function onChange(event: React.ChangeEvent<HTMLInputElement>) {
    setQuery(event.currentTarget.value);
  }

  function onReset() {
    setQuery('');
  }

  function onSubmit() {
    if (items.length > 0) {
      refine(items[0].value);
      setQuery('');
    }
  }

  const mergedTranslations: Required<RefinementListProps['translations']> = {
    resetButtonTitle: 'Clear the search query',
    submitButtonTitle: 'Submit the search query',
    noResultsText: 'No results.',
    showMoreButtonText(options) {
      return options.isShowingMore ? 'Show less' : 'Show more';
    },
    ...translations,
  };

  const uiProps: UiProps = {
    items,
    canRefine,
    onRefine,
    query: inputValue,
    searchBox: searchable && (
      <SearchBoxUiComponent
        inputRef={inputRef}
        placeholder={searchablePlaceholder}
        isSearchStalled={false}
        value={inputValue}
        onChange={onChange}
        onReset={onReset}
        onSubmit={onSubmit}
        translations={{
          submitButtonTitle: mergedTranslations.submitButtonTitle,
          resetButtonTitle: mergedTranslations.resetButtonTitle,
        }}
      />
    ),
    noResults:
      searchable &&
      isFromSearch &&
      items.length === 0 &&
      mergedTranslations.noResultsText,
    canToggleShowMore,
    onToggleShowMore: toggleShowMore,
    isShowingMore,
    translations: {
      showMoreButtonText: mergedTranslations.showMoreButtonText,
    },
  };

  return (
    <RefinementListUiComponent {...props} {...uiProps} showMore={showMore} />
  );
}
