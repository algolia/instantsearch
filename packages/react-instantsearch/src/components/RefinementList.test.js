/* eslint-env jest, jasmine */

import React from 'react';
import renderer from 'react-test-renderer';
import {mount} from 'enzyme';

import RefinementList from './RefinementList';

jest.mock('../widgets/Highlight');

describe('RefinementList', () => {
  it('default refinement list', () => {
    const tree = renderer.create(
      <RefinementList
        refine={() => null}
        createURL={() => '#'}
        items={[
          {label: 'white', value: ['white'], count: 10, isRefined: true},
          {label: 'black', value: ['black'], count: 20, isRefined: false},
          {label: 'blue', value: ['blue'], count: 30, isRefined: false},
        ]}
        limitMin={2}
        limitMax={4}
        showMore={true}
        isFromSearch={false}
        canRefine={true}
      />
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('refinement list with search for facet values but no search results', () => {
    const tree = renderer.create(
      <RefinementList
        refine={() => null}
        searchForFacetValues={() => null}
        createURL={() => '#'}
        items={[
          {label: 'white', value: ['white'], count: 10, isRefined: true},
          {label: 'black', value: ['black'], count: 20, isRefined: false},
          {label: 'blue', value: ['blue'], count: 30, isRefined: false},
        ]}
        isFromSearch={false}
        canRefine={true}
      />
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('refinement list with search for facet values with search results', () => {
    const tree = renderer.create(
      <RefinementList
        refine={() => null}
        searchForFacetValues={() => null}
        createURL={() => '#'}
        items={[
          {label: 'white', value: ['white'], count: 10, isRefined: true, _highlightResult: {label: 'white'}},
        ]}
        isFromSearch={true}
        canRefine={true}
      />
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('applies translations', () => {
    const tree = renderer.create(
      <RefinementList
        refine={() => null}
        searchForFacetValues={() => null}
        createURL={() => '#'}
        items={[
          {label: 'white', value: ['white'], count: 10, isRefined: false},
          {label: 'black', value: ['black'], count: 20, isRefined: false},
          {label: 'blue', value: ['blue'], count: 30, isRefined: false},
        ]}
        isFromSearch={false}
        translations={{
          showMore: ' display more',
          noResults: ' no results',
          placeholder: 'placeholder',
        }}
        canRefine={true}
      />
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('refines its value on change', () => {
    const refine = jest.fn();
    const wrapper = mount(
      <RefinementList
        refine={refine}
        searchForFacetValues={() => null}
        createURL={() => '#'}
        items={[
          {label: 'white', value: ['white'], count: 10, isRefined: false},
          {label: 'black', value: ['black'], count: 20, isRefined: false},
          {label: 'blue', value: ['blue'], count: 30, isRefined: false},
        ]}
        isFromSearch={false}
        canRefine={true}
      />
    );

    const items = wrapper.find('.ais-RefinementList__item');

    expect(items.length).toBe(3);

    const firstItem = items.first().find('.ais-RefinementList__itemCheckbox');

    firstItem.simulate('change', {target: {checked: true}});

    expect(refine.mock.calls.length).toBe(1);
    expect(refine.mock.calls[0][0]).toEqual(['white']);

    wrapper.unmount();
  });

  it('should respect defined limits', () => {
    const refine = jest.fn();
    const wrapper = mount(
      <RefinementList
        refine={refine}
        searchForFacetValues={() => null}
        createURL={() => '#'}
        items={[
          {label: 'white', value: ['white'], count: 10, isRefined: false},
          {label: 'black', value: ['black'], count: 20, isRefined: false},
          {label: 'blue', value: ['blue'], count: 30, isRefined: false},
          {label: 'red', value: ['red'], count: 30, isRefined: false},
          {label: 'green', value: ['green'], count: 30, isRefined: false},
        ]}
        limitMin={2}
        limitMax={4}
        showMore={true}
        isFromSearch={false}
        canRefine={true}
      />
    );

    const items = wrapper.find('.ais-RefinementList__item');

    expect(items.length).toBe(2);

    wrapper.find('.ais-RefinementList__showMore').simulate('click');

    expect(wrapper.find('.ais-RefinementList__item').length).toBe(4);

    wrapper.unmount();
  });

  it('should disabled show more when no more item to display', () => {
    const refine = jest.fn();
    const wrapper = mount(
      <RefinementList
        refine={refine}
        searchForFacetValues={() => null}
        createURL={() => '#'}
        items={[
          {label: 'white', value: ['white'], count: 10, isRefined: false},
          {label: 'black', value: ['black'], count: 20, isRefined: false},
        ]}
        limitMin={2}
        limitMax={4}
        showMore={true}
        isFromSearch={false}
        canRefine={true}
      />
    );

    const items = wrapper.find('.ais-RefinementList__item');

    expect(items.length).toBe(2);

    expect(wrapper.find('.ais-RefinementList__showMoreDisabled')).toBeDefined();

    wrapper.unmount();
  });

  describe('search for facets value', () => {
    const refine = jest.fn();
    const searchForFacetValues = jest.fn();
    const refinementList = <RefinementList
      refine={refine}
      searchForFacetValues={searchForFacetValues}
      createURL={() => '#'}
      items={[
        {
          label: 'white', value: ['white'], count: 10, isRefined: false, _highlightResult: {label: {value: 'white'}},
        },
        {
          label: 'black', value: ['black'], count: 20, isRefined: false, _highlightResult: {label: {value: 'black'}},
        },
      ]}
      isFromSearch={true}
      canRefine={true}
    />;

    it('a searchbox should be displayed if the feature is activated', () => {
      const wrapper = mount(refinementList);

      const searchBox = wrapper.find('.ais-RefinementList__SearchBox');

      expect(searchBox).toBeDefined();

      wrapper.unmount();
    });

    it('searching for a value should call searchForFacetValues', () => {
      const wrapper = mount(refinementList);

      wrapper.find('.ais-RefinementList__SearchBox input').simulate('change', {target: {value: 'query'}});

      expect(searchForFacetValues.mock.calls.length).toBe(1);
      expect(searchForFacetValues.mock.calls[0][0]).toBe('query');

      wrapper.unmount();
    });

    it('should refine the selected value and display selected refinement back', () => {
      const wrapper = mount(refinementList);

      const firstItem = wrapper.find('.ais-RefinementList__item').first().find('.ais-RefinementList__itemCheckbox');
      firstItem.simulate('change', {target: {checked: true}});

      expect(refine.mock.calls.length).toBe(1);
      expect(refine.mock.calls[0][0]).toEqual(['white']);
      expect(wrapper.find('.ais-RefinementList__SearchBox input').props().value).toBe('');

      const selectedRefinements = wrapper.find('.ais-RefinementList__item');
      expect(selectedRefinements.length).toBe(2);

      wrapper.unmount();
    });

    it('hit enter on the search values results list should refine the first facet value', () => {
      refine.mockClear();
      const wrapper = mount(refinementList);

      wrapper.find('form').simulate('submit');

      expect(refine.mock.calls.length).toBe(1);
      expect(refine.mock.calls[0][0]).toEqual(['white']);
      expect(wrapper.find('.ais-RefinementList__SearchBox input').props().value).toBe('');

      const selectedRefinements = wrapper.find('.ais-RefinementList__item');
      expect(selectedRefinements.length).toBe(2);

      wrapper.unmount();
    });
  });

  describe('Panel compatibility', () => {
    it('Should indicate when no more refinement', () => {
      const canRefine = jest.fn();
      const wrapper = mount(
        <RefinementList
          refine={() => null}
          searchForFacetValues={() => null}
          createURL={() => '#'}
          items={[
            {label: 'blue', value: ['blue'], count: 30, isRefined: false},
          ]}
          isFromSearch={false}
          canRefine={true}
        />,
        {
          context: {canRefine},
          childContextTypes: {canRefine: React.PropTypes.func},
        },
      );

      expect(canRefine.mock.calls.length).toBe(1);
      expect(canRefine.mock.calls[0][0]).toEqual(true);
      expect(wrapper.find('.ais-RefinementList__noRefinement').length).toBe(0);

      wrapper.setProps({canRefine: false});

      expect(canRefine.mock.calls.length).toBe(2);
      expect(canRefine.mock.calls[1][0]).toEqual(false);
      expect(wrapper.find('.ais-RefinementList__noRefinement').length).toBe(1);
    });
  });
});
