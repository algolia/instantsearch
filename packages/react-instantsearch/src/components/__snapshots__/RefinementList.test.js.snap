// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RefinementList applies translations 1`] = `
<div>
  <div
    className="ais-RefinementList__root"
  >
    <div
      className="ais-RefinementList__SearchBox"
    >
      <form
        action=""
        className="ais-SearchBox__root"
        noValidate={true}
        onReset={[Function]}
        onSubmit={[Function]}
        role="search"
      >
        <svg
          style={
            Object {
              "display": "none",
            }
          }
          xmlns="http://www.w3.org/2000/svg"
        >
          <symbol
            id="sbx-icon-search-13"
            viewBox="0 0 40 40"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M26.804 29.01c-2.832 2.34-6.465 3.746-10.426 3.746C7.333 32.756 0 25.424 0 16.378 0 7.333 7.333 0 16.378 0c9.046 0 16.378 7.333 16.378 16.378 0 3.96-1.406 7.594-3.746 10.426l10.534 10.534c.607.607.61 1.59-.004 2.202-.61.61-1.597.61-2.202.004L26.804 29.01zm-10.426.627c7.323 0 13.26-5.936 13.26-13.26 0-7.32-5.937-13.257-13.26-13.257C9.056 3.12 3.12 9.056 3.12 16.378c0 7.323 5.936 13.26 13.258 13.26z"
              fillRule="evenodd"
            />
          </symbol>
          <symbol
            id="sbx-icon-clear-3"
            viewBox="0 0 20 20"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M8.114 10L.944 2.83 0 1.885 1.886 0l.943.943L10 8.113l7.17-7.17.944-.943L20 1.886l-.943.943-7.17 7.17 7.17 7.17.943.944L18.114 20l-.943-.943-7.17-7.17-7.17 7.17-.944.943L0 18.114l.943-.943L8.113 10z"
              fillRule="evenodd"
            />
          </symbol>
        </svg>
        <div
          className="ais-SearchBox__wrapper"
          role="search"
        >
          <input
            autoCapitalize="off"
            autoComplete="off"
            autoCorrect="off"
            autoFocus={false}
            className="ais-SearchBox__input"
            onChange={[Function]}
            placeholder="placeholder"
            required={true}
            spellCheck="false"
            type="search"
            value=""
          />
          <button
            className="ais-SearchBox__submit"
            title="Submit your search query."
            type="submit"
          >
            <svg
              role="img"
            >
              <use
                xlinkHref="#sbx-icon-search-13"
              />
            </svg>
          </button>
          <button
            className="ais-SearchBox__reset"
            onClick={[Function]}
            title="Clear the search query."
            type="reset"
          >
            <svg
              role="img"
            >
              <use
                xlinkHref="#sbx-icon-clear-3"
              />
            </svg>
          </button>
        </div>
      </form>
    </div>
    <div
      className="ais-RefinementList__items"
    >
      <div
        className="ais-RefinementList__item"
      >
        <label>
          <input
            checked={false}
            className="ais-RefinementList__itemCheckbox"
            onChange={[Function]}
            type="checkbox"
          />
          <span
            className="ais-RefinementList__itemBox ais-RefinementList__itemBox"
          />
          <span
            className="ais-RefinementList__itemLabel ais-RefinementList__itemLabel"
          >
            white
          </span>
           
          <span
            className="ais-RefinementList__itemCount"
          >
            10
          </span>
        </label>
      </div>
      <div
        className="ais-RefinementList__item"
      >
        <label>
          <input
            checked={false}
            className="ais-RefinementList__itemCheckbox"
            onChange={[Function]}
            type="checkbox"
          />
          <span
            className="ais-RefinementList__itemBox ais-RefinementList__itemBox"
          />
          <span
            className="ais-RefinementList__itemLabel ais-RefinementList__itemLabel"
          >
            black
          </span>
           
          <span
            className="ais-RefinementList__itemCount"
          >
            20
          </span>
        </label>
      </div>
      <div
        className="ais-RefinementList__item"
      >
        <label>
          <input
            checked={false}
            className="ais-RefinementList__itemCheckbox"
            onChange={[Function]}
            type="checkbox"
          />
          <span
            className="ais-RefinementList__itemBox ais-RefinementList__itemBox"
          />
          <span
            className="ais-RefinementList__itemLabel ais-RefinementList__itemLabel"
          >
            blue
          </span>
           
          <span
            className="ais-RefinementList__itemCount"
          >
            30
          </span>
        </label>
      </div>
    </div>
  </div>
</div>
`;

exports[`RefinementList default refinement list 1`] = `
<div>
  <div
    className="ais-RefinementList__root"
  >
    <div
      className="ais-RefinementList__items"
    >
      <div
        className="ais-RefinementList__item ais-RefinementList__itemSelected"
      >
        <label>
          <input
            checked={true}
            className="ais-RefinementList__itemCheckbox ais-RefinementList__itemCheckboxSelected"
            onChange={[Function]}
            type="checkbox"
          />
          <span
            className="ais-RefinementList__itemBox ais-RefinementList__itemBox ais-RefinementList__itemBoxSelected"
          />
          <span
            className="ais-RefinementList__itemLabel ais-RefinementList__itemLabel ais-RefinementList__itemLabelSelected"
          >
            white
          </span>
           
          <span
            className="ais-RefinementList__itemCount ais-RefinementList__itemCountSelected"
          >
            10
          </span>
        </label>
      </div>
      <div
        className="ais-RefinementList__item"
      >
        <label>
          <input
            checked={false}
            className="ais-RefinementList__itemCheckbox"
            onChange={[Function]}
            type="checkbox"
          />
          <span
            className="ais-RefinementList__itemBox ais-RefinementList__itemBox"
          />
          <span
            className="ais-RefinementList__itemLabel ais-RefinementList__itemLabel"
          >
            black
          </span>
           
          <span
            className="ais-RefinementList__itemCount"
          >
            20
          </span>
        </label>
      </div>
    </div>
    <button
      className="ais-RefinementList__showMore"
      disabled={false}
      onClick={[Function]}
    >
      Show more
    </button>
  </div>
</div>
`;

exports[`RefinementList refinement list with search inside items but no search results 1`] = `
<div>
  <div
    className="ais-RefinementList__root"
  >
    <div
      className="ais-RefinementList__SearchBox"
    >
      <form
        action=""
        className="ais-SearchBox__root"
        noValidate={true}
        onReset={[Function]}
        onSubmit={[Function]}
        role="search"
      >
        <svg
          style={
            Object {
              "display": "none",
            }
          }
          xmlns="http://www.w3.org/2000/svg"
        >
          <symbol
            id="sbx-icon-search-13"
            viewBox="0 0 40 40"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M26.804 29.01c-2.832 2.34-6.465 3.746-10.426 3.746C7.333 32.756 0 25.424 0 16.378 0 7.333 7.333 0 16.378 0c9.046 0 16.378 7.333 16.378 16.378 0 3.96-1.406 7.594-3.746 10.426l10.534 10.534c.607.607.61 1.59-.004 2.202-.61.61-1.597.61-2.202.004L26.804 29.01zm-10.426.627c7.323 0 13.26-5.936 13.26-13.26 0-7.32-5.937-13.257-13.26-13.257C9.056 3.12 3.12 9.056 3.12 16.378c0 7.323 5.936 13.26 13.258 13.26z"
              fillRule="evenodd"
            />
          </symbol>
          <symbol
            id="sbx-icon-clear-3"
            viewBox="0 0 20 20"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M8.114 10L.944 2.83 0 1.885 1.886 0l.943.943L10 8.113l7.17-7.17.944-.943L20 1.886l-.943.943-7.17 7.17 7.17 7.17.943.944L18.114 20l-.943-.943-7.17-7.17-7.17 7.17-.944.943L0 18.114l.943-.943L8.113 10z"
              fillRule="evenodd"
            />
          </symbol>
        </svg>
        <div
          className="ais-SearchBox__wrapper"
          role="search"
        >
          <input
            autoCapitalize="off"
            autoComplete="off"
            autoCorrect="off"
            autoFocus={false}
            className="ais-SearchBox__input"
            onChange={[Function]}
            placeholder="Search hereâ€¦"
            required={true}
            spellCheck="false"
            type="search"
            value=""
          />
          <button
            className="ais-SearchBox__submit"
            title="Submit your search query."
            type="submit"
          >
            <svg
              role="img"
            >
              <use
                xlinkHref="#sbx-icon-search-13"
              />
            </svg>
          </button>
          <button
            className="ais-SearchBox__reset"
            onClick={[Function]}
            title="Clear the search query."
            type="reset"
          >
            <svg
              role="img"
            >
              <use
                xlinkHref="#sbx-icon-clear-3"
              />
            </svg>
          </button>
        </div>
      </form>
    </div>
    <div
      className="ais-RefinementList__items"
    >
      <div
        className="ais-RefinementList__item ais-RefinementList__itemSelected"
      >
        <label>
          <input
            checked={true}
            className="ais-RefinementList__itemCheckbox ais-RefinementList__itemCheckboxSelected"
            onChange={[Function]}
            type="checkbox"
          />
          <span
            className="ais-RefinementList__itemBox ais-RefinementList__itemBox ais-RefinementList__itemBoxSelected"
          />
          <span
            className="ais-RefinementList__itemLabel ais-RefinementList__itemLabel ais-RefinementList__itemLabelSelected"
          >
            white
          </span>
           
          <span
            className="ais-RefinementList__itemCount ais-RefinementList__itemCountSelected"
          >
            10
          </span>
        </label>
      </div>
      <div
        className="ais-RefinementList__item"
      >
        <label>
          <input
            checked={false}
            className="ais-RefinementList__itemCheckbox"
            onChange={[Function]}
            type="checkbox"
          />
          <span
            className="ais-RefinementList__itemBox ais-RefinementList__itemBox"
          />
          <span
            className="ais-RefinementList__itemLabel ais-RefinementList__itemLabel"
          >
            black
          </span>
           
          <span
            className="ais-RefinementList__itemCount"
          >
            20
          </span>
        </label>
      </div>
      <div
        className="ais-RefinementList__item"
      >
        <label>
          <input
            checked={false}
            className="ais-RefinementList__itemCheckbox"
            onChange={[Function]}
            type="checkbox"
          />
          <span
            className="ais-RefinementList__itemBox ais-RefinementList__itemBox"
          />
          <span
            className="ais-RefinementList__itemLabel ais-RefinementList__itemLabel"
          >
            blue
          </span>
           
          <span
            className="ais-RefinementList__itemCount"
          >
            30
          </span>
        </label>
      </div>
    </div>
  </div>
</div>
`;

exports[`RefinementList refinement list with search inside items with search results 1`] = `
<div>
  <div
    className="ais-RefinementList__root"
  >
    <div
      className="ais-RefinementList__SearchBox"
    >
      <form
        action=""
        className="ais-SearchBox__root"
        noValidate={true}
        onReset={[Function]}
        onSubmit={[Function]}
        role="search"
      >
        <svg
          style={
            Object {
              "display": "none",
            }
          }
          xmlns="http://www.w3.org/2000/svg"
        >
          <symbol
            id="sbx-icon-search-13"
            viewBox="0 0 40 40"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M26.804 29.01c-2.832 2.34-6.465 3.746-10.426 3.746C7.333 32.756 0 25.424 0 16.378 0 7.333 7.333 0 16.378 0c9.046 0 16.378 7.333 16.378 16.378 0 3.96-1.406 7.594-3.746 10.426l10.534 10.534c.607.607.61 1.59-.004 2.202-.61.61-1.597.61-2.202.004L26.804 29.01zm-10.426.627c7.323 0 13.26-5.936 13.26-13.26 0-7.32-5.937-13.257-13.26-13.257C9.056 3.12 3.12 9.056 3.12 16.378c0 7.323 5.936 13.26 13.258 13.26z"
              fillRule="evenodd"
            />
          </symbol>
          <symbol
            id="sbx-icon-clear-3"
            viewBox="0 0 20 20"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M8.114 10L.944 2.83 0 1.885 1.886 0l.943.943L10 8.113l7.17-7.17.944-.943L20 1.886l-.943.943-7.17 7.17 7.17 7.17.943.944L18.114 20l-.943-.943-7.17-7.17-7.17 7.17-.944.943L0 18.114l.943-.943L8.113 10z"
              fillRule="evenodd"
            />
          </symbol>
        </svg>
        <div
          className="ais-SearchBox__wrapper"
          role="search"
        >
          <input
            autoCapitalize="off"
            autoComplete="off"
            autoCorrect="off"
            autoFocus={false}
            className="ais-SearchBox__input"
            onChange={[Function]}
            placeholder="Search hereâ€¦"
            required={true}
            spellCheck="false"
            type="search"
            value=""
          />
          <button
            className="ais-SearchBox__submit"
            title="Submit your search query."
            type="submit"
          >
            <svg
              role="img"
            >
              <use
                xlinkHref="#sbx-icon-search-13"
              />
            </svg>
          </button>
          <button
            className="ais-SearchBox__reset"
            onClick={[Function]}
            title="Clear the search query."
            type="reset"
          >
            <svg
              role="img"
            >
              <use
                xlinkHref="#sbx-icon-clear-3"
              />
            </svg>
          </button>
        </div>
      </form>
    </div>
    <div
      className="ais-RefinementList__items"
    >
      <div
        className="ais-RefinementList__item ais-RefinementList__itemSelected"
      >
        <label>
          <input
            checked={true}
            className="ais-RefinementList__itemCheckbox ais-RefinementList__itemCheckboxSelected"
            onChange={[Function]}
            type="checkbox"
          />
          <span
            className="ais-RefinementList__itemBox ais-RefinementList__itemBox ais-RefinementList__itemBoxSelected"
          />
          <span
            className="ais-RefinementList__itemLabel ais-RefinementList__itemLabel ais-RefinementList__itemLabelSelected"
          />
           
          <span
            className="ais-RefinementList__itemCount ais-RefinementList__itemCountSelected"
          >
            10
          </span>
        </label>
      </div>
    </div>
  </div>
</div>
`;
